ir/src/boundary_constraints.rs

```diff
@@ -126,7 +126,12 @@ fn validate_expression(
 ) -> Result<(), SemanticError> {
     match expr {
         BoundaryExpr::VectorAccess(vector_access) => {
-            symbol_table.validate_public_input(vector_access.name(), vector_access.idx())
+            let vector_access_type = symbol_table.get_type(vector_access.name())?;
+            symbol_table.validate_vector_access(vector_access, vector_access_type)
```

Comment:
I guess since `validate_vector_access()` it can't really go into `graph.rs` - but even here it feels weird that we first get the type and then ask `SymbolTable` to validate it, even though there is no `SymbolTable`-specific logic in validation code.

I think I'd probably still prefer using `access_vector_element()` and `access_matrix_element()` as it encapsulates the logic more cleanly.

Reply:
Got it. Will change it to that.